ios/Capacitor/Capacitor/WebViewAssetHandler.swift

```diff
@@ -47,67 +49,71 @@ open class WebViewAssetHandler: NSObject, WKURLSchemeHandler {
+                    let parts = rangeString.components(separatedBy: "=")
+                    let streamParts = parts[1].components(separatedBy: "-")
+                    let fromRange = Int(streamParts[0]) ?? 0
+                    var toRange = totalSize - 1
+                    if streamParts.count > 1 {
+                        toRange = Int(streamParts[1]) ?? toRange
+                    }
+                    let rangeLength = toRange - fromRange + 1
+                    try fileHandle.seek(toOffset: UInt64(fromRange))
+                    data = fileHandle.readData(ofLength: rangeLength)
+                    headers["Accept-Ranges"] = "bytes"
+                    headers["Content-Range"] = "bytes \(fromRange)-\(toRange)/\(totalSize)"
+                    headers["Content-Length"] = String(data.count)
+                    let response = HTTPURLResponse(url: localUrl, statusCode: 206, httpVersion: nil, headerFields: headers)
+                    urlSchemeTask.didReceive(response!)
+                    try fileHandle.close()
+                } else {
+                    if !stringToLoad.contains("cordova.js") {
+                        if isMediaExtension(pathExtension: url.pathExtension) {
+                            data = try Data(contentsOf: fileUrl, options: Data.ReadingOptions.mappedIfSafe)
+                        } else {
+                            data = try Data(contentsOf: fileUrl)
+                        }
+                    }
+                    let urlResponse = URLResponse(url: localUrl, mimeType: mimeType, expectedContentLength: data.count, textEncodingName: nil)
+                    let httpResponse = HTTPURLResponse(url: localUrl, statusCode: 200, httpVersion: nil, headerFields: headers)
                     if isMediaExtension(pathExtension: url.pathExtension) {
-                        data = try Data(contentsOf: fileUrl, options: Data.ReadingOptions.mappedIfSafe)
+                        urlSchemeTask.didReceive(urlResponse)
                     } else {
-                        data = try Data(contentsOf: fileUrl)
+                        urlSchemeTask.didReceive(httpResponse!)
                     }
                 }
-                let urlResponse = URLResponse(url: localUrl, mimeType: mimeType, expectedContentLength: data.count, textEncodingName: nil)
-                let httpResponse = HTTPURLResponse(url: localUrl, statusCode: 200, httpVersion: nil, headerFields: headers)
-                if isMediaExtension(pathExtension: url.pathExtension) {
-                    urlSchemeTask.didReceive(urlResponse)
-                } else {
-                    urlSchemeTask.didReceive(httpResponse!)
-                }
+                urlSchemeTask.didReceive(data)
+            } catch let error as NSError {
+                urlSchemeTask.didFailWithError(error)
+                self.pendingTasks.removeAll(where: { $0 == urlSchemeTask.hash })
```

Comment:
When this is updated to be a Set, this would just be changed to be `pendingTasks.remove(urlSchemeTask.hash)`